clients/hadoopfs/src/main/java/io/lakefs/BulkDeleter.java

```diff
@@ -0,0 +1,158 @@
+package io.lakefs;
+
+import io.lakefs.clients.api.ApiException;
+import io.lakefs.clients.api.model.ObjectErrorList;
+import io.lakefs.clients.api.model.PathList;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+
+class BulkDeleter implements Closeable {
+    private static final int defaultBulkSize = 1000;
+
+    private final ExecutorService executor;
+    private final Callback callback;
+    private final String repository;
+    private final String branch;
+    private final int bulkSize;
+
+    private PathList pathList;
+    // TODO(ariels): Configure this!
+    private final int concurrency = 5;
+    private Queue<Future<ObjectErrorList>> deletions = new ArrayDeque();
+
+    public static interface Callback {
```

Comment:
why static?

Reply:
It's static because it is not inner AFAIK.  I realize that Java probably ends up building the class of an interface as non-inner even without this keyword... but that is even more confusing to me.

Reply:
Don't think it's needed, but it doesn't matter too much...